### PR TITLE
[ML-DataFrame] refactor pivot to only take the pivot config

### PR DESCRIPTION
**Pull request by hendrikmuhs**
refactor pivot class to only take the config at construction, other parameters are passed in as part of method that require them

### CODE REVIEW

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/2

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java [hunk 1]
```diff
@@ -180,7 +180,16 @@ protected void onFinish(ActionListener<Void> listener) {
 
     @Override
     protected SearchRequest buildSearchRequest() {
-        return pivot.buildSearchRequest(getPosition(), pageSize);
+        SearchRequest searchRequest = new SearchRequest(getConfig().getSource().getIndex());
```
Comment (davidkyle) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java]: This code looks to functionally the same as 
`Pivot.buildSearchRequest(getConfig().getSource(), getPosition, pageSize)`

Comment (hendrikmuhs) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java]: yes, in this version. It is/will be fundamentally different for "continuous". 

I will re-check, you are probably right, in this branch I can call `pivot.build...`

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**